File: lc/c/069.c
struct ListNode* reverseBetween(struct ListNode* head, int left, int right){
    struct ListNode dummy = {.next = head}, *front = &dummy;
    for (int i = 0; i < left-1; ++i)
        front = front->next;

    struct ListNode *end = front->next;
    for (int i = 0; i < right-left; ++i){
        struct ListNode *tmp1 = front->next;
        struct ListNode *tmp2 = end->next->next;
        front->next = end->next;
        front->next->next = tmp1;
        end->next = tmp2;
    }
    return dummy.next;
}

File: lc/py/069.py
class Solution:
    def reverseBetween(self, head: ListNode, left: int, right: int) -> ListNode:
        front = dummy = ListNode(0, head)
        for _ in range(left-1):
            front = front.next

        end = front.next
        for _ in range(right-left):
            tmp = end.next
            front.next, tmp.next, end.next = tmp, front.next, tmp.next

        return dummy.next
